common/thrifthelpers/thrift_serializer.go

```diff
@@ -0,0 +1,40 @@
+package thrifthelpers
+
+import "github.com/apache/thrift/lib/go/thrift"
+
+type Serializer interface {
```

Comment:
so in order to have an object that implements this interface for Json & Binary, you'll need to declare a struct and have Serialize and Deserialize be methods on it.  

type JsonSerializer struct {}
func (j JsonSerializer) Serialize ...
func (j JsonSerializer) Serialize ...

It looks like you do something like this in job_def_serializer.  Is this the piece that needs to be refactored next you mentioned?